### PR TITLE
Snapping average perceptron and OGD to the other constructors.

### LINKED ISSUE #910: Fix AP and OGD metadata propagation and behavior
In the process of writing #909 I discovered that AP is now seemingly uncalibrated, and cannot accept float labels, and OGD throws at training. 

Enable the `OnlineLinearWorkout` test in #909 and fix the resulting bugs.

### PR DESCRIPTION
**Pull request by sfilipi**
Addresses part of #910. 
Won't close #910 as part of this PR, to deal with the calibration in AP separately.

### CODE REVIEW

File: test/Microsoft.ML.StaticPipelineTesting/Training.cs [hunk 1]
```diff
@@ -629,8 +629,8 @@ public void OnlineGradientDescent()
             var loss = new SquaredLoss();
 
             var est = reader.MakeNewEstimator()
-                .Append(r => (r.label, score: ctx.Trainers.OnlineGradientDescent(r.label, r.features,
-                // lossFunction:loss,
+                .Append(r => (r.label, score: ctx.Trainers.OnlineGradientDescent(r.label, r.features,
```
Comment (Zruty0) [inline test/Microsoft.ML.StaticPipelineTesting/Training.cs]: >es,  [](start = 98, length = 4)

space

File: test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs [hunk 2]
```diff
@@ -632,7 +632,7 @@ private void MixMatch(string dataPath)
             IEstimator<ITransformer> dynamicPipe = learningPipeline.AsDynamic;
 
             // Create a binary classification trainer.
-            var binaryTrainer = new AveragedPerceptronTrainer(env, new AveragedPerceptronTrainer.Arguments());
+            var binaryTrainer = new AveragedPerceptronTrainer(env, "Label", "Features");
```
Comment (Zruty0) [inline test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs]: >"Label", "Features" [](start = 67, length = 19)

please make the corresponding update in the markdown file
